scripts/build.sh

```diff
@@ -3,110 +3,187 @@
+# via a non secure channel. (ECDH)
+#
+#############################################
+ID_PRIV_KEY='./certs/broker-identity.private.pem'
+ID_PUB_KEY='./certs/broker-identity.public.pem'
+
+if [[ -f "$ID_PRIV_KEY" ]]; then
+  echo "WARNING: ID already exists for Broker Daemon. Skipping ID generation"
+elif [[ -f "$ID_PUB_KEY" ]]; then
+  echo "WARNING: ID Public Key already exists for Broker Daemon. Skipping ID generation"
+elif [ "$NO_IDENTITY" != "true" ]; then
+  mkdir -p ./certs
+  openssl ecparam -name prime256v1 -genkey -noout -out $ID_PRIV_KEY
+  openssl ec -in $ID_PRIV_KEY -pubout -out $ID_PUB_KEY
+fi
+
+if [ "$NO_DOCKER" == "false" ]; then
+  # NOTE: The names specified with `-t` directly map to the service names in
+  # the applicable services docker-compose file
+  echo "Building broker docker images"
+  docker build -t sparkswap_sparkswapd -f ./docker/sparkswapd/Dockerfile ./
+fi
+
+if [ -f docker-compose.override.yml ]; then
   # Let the user know that an override file exists which may mean that the user
   # will have settings they do not expect
-  echo "A 'docker-compose.override.yml' file exists, but you are not using a supported"
-  echo "environment: local or regtest"
   echo ""
+  echo "WARNING: A 'docker-compose.override.yml' file exists"
   echo "WARNING: This may add unwanted settings to the broker that could effect how your daemon runs."
   echo ""
 fi
 
-echo "Generating certificates for RPC"
-
-# Generate certs for the CLI
-rm -rf ./certs
-mkdir -p ./certs
-
-# Set paths for self-signed key pair
-KEY_PATH="./certs/broker-rpc-tls.key"
-CERT_PATH="./certs/broker-rpc-tls.cert"
-CSR_PATH="./certs/broker-rpc-csr.csr"
-EXTERNAL_ADDRESS=${EXTERNAL_ADDRESS:-localhost}
-
-openssl ecparam -genkey -name prime256v1 -out $KEY_PATH
-
-openssl req -new -sha256 \
-  -reqexts SAN \
-  -extensions SAN \
-  -config <(cat /etc/ssl/openssl.cnf \
-      <(printf "\n[SAN]\nsubjectAltName=DNS:${EXTERNAL_ADDRESS},DNS:localhost")) \
-  -key $KEY_PATH \
-  -out $CSR_PATH \
-  -subj "/CN=$EXTERNAL_ADDRESS/O=sparkswap"
-
-openssl req -x509 -sha256 \
-  -reqexts SAN \
-  -extensions SAN \
-  -config <(cat /etc/ssl/openssl.cnf \
-      <(printf "\n[SAN]\nsubjectAltName=DNS:${EXTERNAL_ADDRESS},DNS:localhost")) \
-  -days 36500 \
-  -key $KEY_PATH \
-  -in $CSR_PATH \
-  -out $CERT_PATH
-
-rm $CSR_PATH
-
-echo "Building broker docker images"
-KEY_PATH=$KEY_PATH CERT_PATH=$CERT_PATH npm run build-images
-
 # We can skip the copying of certs to a local directory if the current build is
 # a standalone broker
-if [ "$ARG" == "no-cli" ]; then
-  rm -rf ./certs
+if [ "$NO_CLI" == "true" ] || [ "$NO_CERTS" == "true" ]; then
   exit 0
 fi
 
-echo "Copying certs"
-
 echo "Making local ~/.sparkswap certs directory"
 DIRECTORY=~/.sparkswap
 mkdir -p $DIRECTORY/certs
 
 echo "Copying certs to local certs directory"
+# Not sure what to do here because the certs is gonna fuck everything up
```

Comment:
probably not a comment we want in the codebase

Reply:
woops, that is my bad. I will rebase